test/unit/test_quilt.py

```diff
@@ -136,29 +146,29 @@ def test_memory_reference_parameter():
     assert _match("DEFCAL RX(%theta) q", "DECLARE theta REAL\nRZ(theta) 0") is None
 
 
-# TODO: Calibration API: match calibration
 def test_measure_calibration_match():
     matches = [
-        ("DEFCAL MEASURE 0", "MEASURE 0"),
-        ("DEFCAL MEASURE q", "MEASURE 0"),
+        ("DEFCAL MEASURE 0 dest", "MEASURE 0"),
+        ("DEFCAL MEASURE q dest", "MEASURE 0"),
         ("DEFCAL MEASURE 0 b", "DECLARE ro BIT\nMEASURE 0 ro[0]"),
         ("DEFCAL MEASURE q b", "DECLARE ro BIT\nMEASURE 1 ro[0]"),
     ]
 
     for cal, instr in matches:
-        assert _match(cal, instr) is not None
+        assert _match_measure(cal, instr) is not None
 
-    assert _match(cal, instr).settings[FormalArgument("q")] == Qubit(1)
-    assert _match(cal, instr).settings[FormalArgument("b")] == MemoryReference("ro")
+    match = _match_measure(cal, instr)
+    assert match.settings[FormalArgument("q")] == Qubit(1)
+    assert match.settings[MemoryReference("b")] == MemoryReference("ro")
 
     mismatches = [
-        ("DEFCAL MEASURE 1", "MEASURE 0"),
-        ("DEFCAL MEASURE 0 b", "MEASURE 0"),
-        ("DEFCAL MEASURE q", "DECLARE ro BIT\nMEASURE 0 ro[0]"),
+        ("DEFCAL MEASURE 1 dest", "MEASURE 0"),
+        # ("DEFCAL MEASURE 0 b", "MEASURE 0"),
+        # ("DEFCAL MEASURE q dest", "DECLARE ro BIT\nMEASURE 0 ro[0]"),
```

Comment:
All of the calibration tests pass except for these commented out cases.

Starting with this one:
```py
("DEFCAL MEASURE 0 b", "MEASURE 0")
```

I'm assuming it shouldn't match, because there is no `DECLARE` for `b`. This could be solved in `quil_rs` by comparing `Measurement.target` and `MeasureCalibrationDefinition.parameter` but the former is a `MemoryReference` and the latter is a string. Maybe we could compare the name of the reference, but that seems fragile. Should MeasureCalibrationDefinition just use a `MemoryReference` for it's parameter? This is closer to what `pyQuil` does, but `pyQuil` allows it to be optional.

Moving on to this:
```
("DEFCAL MEASURE q dest", "DECLARE ro BIT\nMEASURE 0 ro[0]")
```
In the original test, `dest` was committed because pyQuil allows a destination to be optional. If we move towards _requiring_ a destination, then I think this case is irrelevant.

Reply:
In the interest of moving fast, let's make the destination non-optional. We can do the string comparison or change `quil-rs` to expect a memory reference there, in anticipation of a change to the spec.

Reply:
I've done so and marked the commit as a BREAKING CHANGE so we can keep track of it